apps/src/code-studio/components/progress/UnitCalendarDialog.jsx

```diff
@@ -0,0 +1,127 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import i18n from '@cdo/locale';
+import BaseDialog from '@cdo/apps/templates/BaseDialog';
+import Button from '@cdo/apps/templates/Button';
+import UnitCalendar from './UnitCalendar';
+import _ from 'lodash';
+import {unitCalendarLesson} from '@cdo/apps/templates/progress/unitCalendarLessonShapes';
+
+const styles = {
+  dialog: {
+    textAlign: 'left',
+    paddingLeft: 20,
+    paddingRight: 20,
+    paddingBottom: 20
+  },
+  button: {
+    float: 'right',
+    marginTop: 30
+  }
+};
+
+export default class UnitCalendarDialog extends Component {
+  static propTypes = {
+    isOpen: PropTypes.bool,
+    handleClose: PropTypes.func,
+    lessons: PropTypes.arrayOf(unitCalendarLesson).isRequired,
+    weeklyInstructionalMinutes: PropTypes.number.isRequired
+  };
+
+  generateSchedule = () => {
```

Comment:
If this schedule is generated here and then just passed to UnitCalendar without using any of it in this component I think we should move the calculation into UnitCalendar